src/hbase/test-util/mini-cluster.cc

```diff
@@ -306,3 +289,14 @@ jclass hbase::MiniCluster::FindClassAndGetGlobalRef(const char *cls_name) {
   env_->DeleteLocalRef(local_ref);
   return static_cast<jclass>(global_ref);
 }
+
+void hbase::MiniCluster::StopCluster() {
+  CHECK_NOTNULL(env_);
+  lock_guard<mutex> lock(lock_);
```

Comment:
Unless I'm missing something ( usually am ), If we move assignment of testing_util_class to the constructor we shouldn't need a mutex and we shouldn't need to check if htu exists. This would allow StopCluster to be used in the destructor safely.

Reply:
Ya, I actually thought about this, the issue is that the cluster operations could be a little more complex than just start/stop of cluster. A typical sequence (in the future tests) could be like

StartCluster()
StopCluster()
StartCluster() ...again
AddRegionServer()
StopRegionServer()
Stop()
.....

So typically all of them need to grab this mutex before making the change, otherwise we'd probably end up in a weird state (for ex: destroying the JVM while say addRegionServer() etc.)... So I retained the mutex as is rather than moving it into the c'tor and d'tor. What do you think?

Reply:
Sorry for the delay. I've given this some thought and agree with what you are asserting. It would hinder the flow of logic and/or require odd levels of scoping in that type of sequence. 

Scope guards might help as would a singleton around the initialization of the JVM; however, I think what is here is fine. I don't think you should re-do it solves the issue. Thanks!